File: testdata/bench/csharp/BenchmarkTest00016.cs
using System.Data.SqlClient;
using Microsoft.AspNetCore.Mvc;

namespace BenchApp.Controllers
{
    public class SqliSafeController : Controller
    {
        [HttpGet]
        public IActionResult GetById()
        {
            string rawId = Request.Query["id"];
            int id = int.Parse(rawId);
            var conn = new SqlConnection("Server=localhost;Database=app;");
            conn.Open();
            var cmd = new SqlCommand("SELECT * FROM users WHERE id = @id", conn);
            cmd.Parameters.AddWithValue("@id", id);
            return Ok();
        }
    }
}

File: testdata/bench/swift/BenchmarkTest00016.swift
import Vapor
import SQLite3

func handler016(_ req: Request) throws -> String {
    let rawId = try req.parameters.require("id")
    guard let numericId = Int(rawId) else { throw Abort(.badRequest) }
    var db: OpaquePointer?
    sqlite3_open(":memory:", &db)
    sqlite3_exec(db, "SELECT * FROM users WHERE id = \(numericId)", nil, nil, nil)
    return "ok"
}
